fault_injection/fi_gear/husky/husky_vcc.py

```diff
@@ -0,0 +1,101 @@
+        trigger_delay_min: Lower bound for the trigger delay.
+        trigger_delay_max: Upper bound for the trigger delay.
+        trig_step: Step for the trigger delay.
+    """
+    def __init__(self, pll_frequency: int, glitch_width_min: float,
+                 glitch_width_max: float, glitch_width_step: float,
+                 trigger_delay_min: int, trigger_delay_max: int, trig_step: int):
+        # Set Husky parameters.
+        self.scope = None
+        self.pll_frequency = pll_frequency
+
+        # Set glitch parameter space.
+        self.glitch_width_min = glitch_width_min
+        self.glitch_width_max = glitch_width_max
+        self.glitch_width_step = glitch_width_step
+        self.trigger_delay_min = trigger_delay_min
+        self.trigger_delay_max = trigger_delay_max
+        self.trig_step = trig_step
+
+        # Init husky.
+        self.init_husky()
+
+    def init_husky(self) -> None:
+        """ Initialize husky.
+
+        Configure Husky crowbar in such a way that the glitcher uses the HP
+        MOSFET and a single glitch is pll_frequency long.
+        """
+        check_version.check_husky("1.5.0")
+        self.scope = cw.scope()
+
+        if not self.scope._is_husky:
+            raise RuntimeError("Only ChipWhisperer Husky is supported!")
+        # Configure the clock.
+        self.scope.clock.clkgen_freq = self.pll_frequency
+        # Initialize the voltage glitching.
+        self.scope.vglitch_setup('hp', default_setup=False)
+        # Glitch output is high for glitch.repeat cycles. One cylce is
+        # clkgen_freq.
+        self.scope.glitch.output = "enable_only"
+
+        # Configure trigger.
+        self.scope.glitch.trigger_src = "ext_single"
+        self.scope.trigger.triggers = "tio4"
+
+    def arm_trigger(self, fault_parameters: dict) -> None:
+        """ Arm the trigger.
+
+        Configures the glitcher to inject a fault of width glitch.repeat cycles
+        after the trigger delay.
+
+        Args:
+            A dict containing the FI parameters.
+        """
+        self.scope.glitch.repeat = fault_parameters['glitch_width']
+        self.scope.glitch.ext_offset = fault_parameters['trigger_delay']
+        self.scope.arm()
+
+    def generate_fi_parameters(self) -> dict:
+        """ Generate random voltage glitch parameters within the provided
+            limits.
+
+        Returns:
+            A dict containing the FI parameters.
+        """
+        parameters = {}
+        parameters["glitch_width"] = random_float_range(self.glitch_width_min,
+                                                        self.glitch_width_max,
+                                                        self.glitch_width_step)
+        parameters["trigger_delay"] = random_float_range(self.trigger_delay_min,
+                                                         self.trigger_delay_max,
+                                                         self.trig_step)
```

Comment:
I am wondering whether there is a smart way to avoid re-running the experiment using the same parameters. May keep them all around in a big array and generate new values until the combination doesn't completely match anymore?

Reply:
Also, I think we discussed that some parameters like the x,y position maybe doesn't need to be random. Obviously, we don't have this for Husky but maybe we should already add support for non-randomized parameters?

Reply:
Yeah, this could be an option. However, in my experience, a completely random strategy works also quite well.

Regarding the deterministic fault parameter sweep, I'll create an issue & implement it.